### PR TITLE
Fix adapter's error on taobao platform

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by minggo** [comment]
What's the error?

File: platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/localStorage.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ const localStorage = {
   },
 
   removeItem(key) {
-    my.removeStorageSync(key)
+    my.removeStorageSync({ key })
```
Comment (bofeng-song) [inline platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/localStorage.js:29]: An object parameter is required by API

File: pal/wasm/wasm-minigame.ts [hunk 2]
```diff
@@ -119,11 +119,7 @@ function getPlatformBinaryUrl (binaryUrl: string): Promise<string> {
         if (XIAOMI) {
             resolve(`src/cocos-js/${binaryUrl}`);
         } if (TAOBAO_MINIGAME && WASM_SUBPACKAGE) {
-            if (minigame.isDevTool) {
-                resolve(`cocos-js/${binaryUrl}`);
-            } else {
-                resolve(`__ccWasmAssetSubpkg__/${basename(binaryUrl)}`);
-            }
+            resolve(`__ccWasmAssetSubpkg__/${basename(binaryUrl)}`);
```
Comment (bofeng-song) [inline pal/wasm/wasm-minigame.ts:122]: IDE tools run the same logic now. No need to differentiate between the development and real device environments.

**Comment by bofeng-song** [comment]
> What's the error?

This will cause removeStorageSync to not work properly, and no error will be reported.

**Review by minggo — APPROVED** [review]

**Pull request merged**